e2e/rook-test-framework/rook-infra-manager/rook_test_infra_manager.go

```diff
@@ -0,0 +1,323 @@
+		panic(err)
+	}
+
+	//Drain node 2
+	_, _, err = k8sClient.ExecuteCmd([]string{"drain", "kube-node-2", "--force", "--ignore-daemonsets"})
+
+	if err != nil && !strings.EqualFold(err.Error(), k8sFalsePostiveSuccessErrorMsg) {
+		panic(err)
+	}
+
+	//Delete 2nd unnecessary node
+	_, _, err = k8sClient.ExecuteCmd([]string{"delete", "node", "kube-node-2", "--force"})
+
+	if err != nil && !strings.EqualFold(err.Error(), k8sFalsePostiveSuccessErrorMsg) {
+		panic(err)
+	}
+
+	//if any of these methods fail, the executeDockerCommand will panic
+	_, k8sMasterContainerId := r.executeDockerCommand("", enums.Ps, "--filter", "name=kube-master", "--format", "{{.ID}}")
+
+	//Install ceph-commons on each k8s node
+	r.executeDockerCommand(k8sMasterContainerId, enums.Exec,"bin/bash", "-c", "apt-get -y update && apt-get install -qqy ceph-common")
+
+	_, k8sNode1ContainerId := r.executeDockerCommand("", enums.Ps, "--filter", "name=kube-node-1", "--format", "{{.ID}}")
+
+	r.executeDockerCommand(k8sNode1ContainerId, enums.Exec,"bin/bash", "-c", "apt-get -qy update && apt-get install -qqy ceph-common")
+}
+
+//method for create rook-operator via kubectl
+func createK8sRookOperator(k8sHelper *utils.K8sHelper, tag string) error {
+	raw, err := ioutil.ReadFile(podSpecPath + "/" + rookOperatorFileName)
+
+	if err != nil {
+		return err
+	}
+
+	rawUpdated := bytes.Replace(raw, []byte(rookOperatorImagePodSpecTag), []byte(tag), 1)
+	rookOperator := string(rawUpdated)
+
+	_, _, exitCode := r.transportClient.CreateWithStdin(rookOperator)
+
+	if exitCode != 0 {
+		return errors.New(string("Failed to create rook-operator pod; kubectl exit code = " + string(exitCode)))
+	}
+
+	fmt.Println()
+	if !k8sHelper.IsThirdPartyResourcePresent(rookOperatorCreatedTpr) {
+		return errors.New ("Failed to start Rook Operator; k8s thirdpartyresource did not appear")
+	} else {
+		fmt.Println("Rook Operator started")
+	}
+
+	return nil
+}
+
+func createk8sRookCluster(k8sHelper *utils.K8sHelper) error {
+	raw, err := ioutil.ReadFile(podSpecPath + "/" + rookClusterFileName)
+
+	if err != nil {
+		return err
+	}
+
+	rookCluster := string(raw)
+
+	_, _, exitCode := r.transportClient.CreateWithStdin(rookCluster)
+
+	if exitCode != 0 {
+		return errors.New("Failed to create rook-cluster pod; kubectl exit code = " + string(exitCode))
+	}
+
+	if !k8sHelper.IsServiceUpInNameSpace("rook-api") {
+		fmt.Println("Rook Cluster couldn't start")
+	} else {
+		fmt.Println("Rook Cluster started")
+	}
+
+	return nil
+}
+
+func (r *rookTestInfraManager) InstallRook(tag string) (err error, client contracts.Irook_client) {
+	//Create rook operator
+	k8sHelp := utils.CreatK8sHelper()
+
+	err = createK8sRookOperator(k8sHelp, tag)
+
+	if err != nil {
+		panic(err)
+	}
+
+	time.Sleep(10 * time.Second)	///TODO: add real check here
+
+	//Create rook cluster
+	err = createk8sRookCluster(k8sHelp)
+
+	if err != nil {
+		panic(err)
+	}
+
+	time.Sleep(5 * time.Second)
```

Comment:
This may be incorrect, we are checking for the resource, but not checking for a status of the resource when found... ill add code for this...